paddle/fluid/imperative/layer.cc

```diff
@@ -311,6 +312,42 @@ void VarBase::CopyFrom(const VarBase& src, const bool blocking) {
   }
 }
 
+void VarBase::MoveTo(const platform::Place& dst_place, bool blocking) {
+  PADDLE_ENFORCE_EQ(
+      Var().IsInitialized() && (Var().IsType<framework::LoDTensor>() ||
+                                Var().IsType<framework::SelectedRows>()),
+      true, platform::errors::InvalidArgument(
+                "Variable is not initialized or Variable's type is not "
+                "LoDTensor or SelectedRows when moving to destination place"));
+
+  if (platform::is_same_place(Place(), dst_place)) {
+    // do nothing
+    return;
+  }
+
+  const framework::Tensor* src_tensor = nullptr;
+  if (Var().IsType<framework::LoDTensor>()) {
+    src_tensor = &Var().Get<framework::LoDTensor>();
+  } else {
+    src_tensor = &Var().Get<framework::SelectedRows>().value();
+  }
+
+  framework::Tensor dst_tensor;
+  dst_tensor.Resize(src_tensor->dims());
+  dst_tensor.mutable_data(dst_place, src_tensor->type());
```

Comment:
Does the difference between `dst_tensor` and `src_tensor`  **only** is **the place**? 
If yes,  
1. `dst_tensor` need to share the `inplace_version_counter_` with `src_tensor` to ensure the correctness of gradient calculation
Please add code like:
```
dst_tensor->ShareInplaceVersionCounterWith(src_tensor);
```

2. Here only set the same `dims` and `type` as `src_tensor`, does it ensure that the other attributes of `src_tensor` and `dst_tensor` are consistent? such as `offset_`, `layout_`?

Reply:
Thanks to find this, I will fix this.

Reply:
use `src_tensor->ShareBufferWith(dst_tensor)` to reset `holder_` and `offset_`. Other information of src_tensor is still keep unchange.